research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+
+The `delegatesfocus` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that when the `popup` element is shown, focus should move to the first
+focusable descendent of the `popup` element.
+
+**Example:**
+
+```html
+<popup delegatesfocus>
+  <p>I am not a focusable element.</p>
+  <p>Nor am I.</p>
+  <button>I will be focused whenever the popup becomes focused.</button>
+</popup>
+```
+
+### Focusing steps
+
+The popup [focusing steps](https://html.spec.whatwg.org/multipage/interaction.html#focusing-steps)
+for a `popup` element _subject_ are as follows:
+
+1. If the `delegatesfocus` attribute is specified on _subject_, let _control_ be the first focusable
+   descendant element of _subject_, in [tree order](https://dom.spec.whatwg.org/#concept-tree-order),
+   that is not inert and has the
+   [autofocus](https://html.spec.whatwg.org/multipage/interaction.html#attr-fe-autofocus) attribute
+   specified.
+2. If there is no such descendent, but `delegatesfocus` is specified, let _control_ be _subject_.
+3. Otherwise, if the `autofocus` attribute is specified on _subject_, let _control_ be _subject_.
+4. Otherwise, return.
+5. Run the [focusing steps](https://html.spec.whatwg.org/multipage/interaction.html#focusing-steps)
+   for _control_.
+6. Let topDocument be the
+   [active document](https://html.spec.whatwg.org/multipage/browsers.html#active-document) of
+   _control_'s [node document](https://dom.spec.whatwg.org/#concept-node-document)'s
+   [browsing context](https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc)'s
+   [top-level browsing context](https://html.spec.whatwg.org/multipage/browsers.html#top-level-browsing-context).
+7. If control's [node document](https://dom.spec.whatwg.org/#concept-node-document)'s
+   [origin](https://dom.spec.whatwg.org/#concept-document-origin) is not the
+   [same](https://html.spec.whatwg.org/multipage/origin.html#same-origin) as the
+   [origin](https://dom.spec.whatwg.org/#concept-document-origin) of _topDocument_, then return.
+8. [Empty](https://infra.spec.whatwg.org/#list-empty) _topDocument_'s
+   [autofocus candidates](https://html.spec.whatwg.org/multipage/interaction.html#autofocus-candidates).
+9. Set _topDocument_'s
+   [autofocus processed flag](https://html.spec.whatwg.org/multipage/interaction.html#autofocus-processed-flag)
+   to true.
+
+<p class="note">
+  Without the precense of the autofocus or delegatefocus attribute, focus remains on the active
+  element. This behavior is to enable scenarios where the popup is used in a composite control. For
+  example, a combobox where the user expects their focus to stay in the text input instead of moving
+  automatically to the listbox popup as and when it appears.
+</p>
+
+<p class="question">
+  These focusing steps effectively assert that the delegatesfocus attribute takes precedence over
```

Comment:
That's also how I implemented it, and it made sense to me at the time. For whatever that's worth! 😆

Reply:
Huzzah!